File: crashlytics-slf4j/src/main/java/de/lemona/android/crashlytics/CrashlyticsLoggerFactory.java
package de.lemona.android.crashlytics;

import org.slf4j.ILoggerFactory;
import org.slf4j.Logger;

import java.util.concurrent.ConcurrentHashMap;

import de.lemona.android.slf4j.AndroidLogger;

public class CrashlyticsLoggerFactory implements ILoggerFactory {

    private final ConcurrentHashMap<String, CrashlyticsLogger> loggers = new ConcurrentHashMap<>();

    @Override
    public Logger getLogger(String name) {
        final CrashlyticsLogger existing = loggers.get(name);
        if (existing != null) return existing;
        final CrashlyticsLogger created = new CrashlyticsLogger(name);
        final CrashlyticsLogger previous = loggers.putIfAbsent(name, created);
        if (previous != null) return previous;
        return created;
    }

}

File: crashlytics-slf4j/src/main/java/de/lemona/android/crashlytics/CrashlyticsLoggerFactory.kt
package de.lemona.android.crashlytics

import org.slf4j.ILoggerFactory
import de.lemona.android.crashlytics.CrashlyticsLogger
import org.slf4j.Logger
import java.util.concurrent.ConcurrentHashMap

class CrashlyticsLoggerFactory : ILoggerFactory {
    private val loggers = ConcurrentHashMap<String, CrashlyticsLogger>()
    override fun getLogger(name: String): Logger {
        val existing = loggers[name]
        if (existing != null) return existing
        val created = CrashlyticsLogger(name)
        val previous = loggers.putIfAbsent(name, created)
        return previous ?: created
    }
}